tools/azsdk-cli/update-mcp-tools-docs.ps1

```diff
@@ -0,0 +1,137 @@
+    
+    $psi = New-Object System.Diagnostics.ProcessStartInfo
+    $psi.FileName = $ExePath
+    $psi.Arguments = "start"
+    $psi.UseShellExecute = $false
+    $psi.RedirectStandardInput = $true
+    $psi.RedirectStandardOutput = $true
+    $psi.RedirectStandardError = $true
+    $psi.CreateNoWindow = $true
+    
+    $process = New-Object System.Diagnostics.Process
+    $process.StartInfo = $psi
+    [void]$process.Start()
+    
+    Start-Sleep -Seconds 3
+    
+    if ($process.HasExited) {
+        throw "MCP server exited with code: $($process.ExitCode)"
+    }
+    
+    return $process
+}
+
+function Get-McpTools {
+    param([System.Diagnostics.Process]$Process)
+    
+    $request = @{
+        jsonrpc = "2.0"
+        id = 1
+        method = "tools/list"
+        params = @{}
+    } | ConvertTo-Json -Compress
+    
+    $Process.StandardInput.WriteLine($request)
+    $Process.StandardInput.Flush()
+    
+    Start-Sleep -Seconds 2
+    $response = $Process.StandardOutput.ReadLine()
+    
+    if (-not $response -or $response -notmatch '^\{.*"jsonrpc".*\}$') {
+        throw "No valid JSON-RPC response received"
+    }
+    
+    $responseObj = $response | ConvertFrom-Json
+    
+    if (-not $responseObj.result -or -not $responseObj.result.tools) {
+        throw "Invalid response format"
+    }
+    
+    return $responseObj.result.tools
+}
+
+function New-ToolsMarkdown {
+    param(
+        [array]$Tools,
+        [string]$Version
+    )
+    
+    $content = @"
+# Tools available in Azure SDK MCP server
+
+This document provides a comprehensive list of all MCP (Model Context Protocol) tools supported by the Azure SDK MCP server version $Version.
+
+## Tools list
+
+| Name | Description |
+|----------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------|
+"@
+    
+    foreach ($tool in $Tools | Sort-Object name) {
+        $name = $tool.name
+        $description = ($tool.description -replace "\r?\n", " ") -replace "\s+", " "
+        $content += "`n| $name | $description |"
+    }
+    
+    return $content
+}
+
+function Get-Version {
+    param([string]$CsprojPath)
+    
+    if (Test-Path $CsprojPath) {
+        [xml]$csproj = Get-Content $CsprojPath
+        $versionPrefix = $csproj.Project.PropertyGroup.VersionPrefix
+        if ($versionPrefix) {
+            return $versionPrefix
+        }
+    }
+    return "latest"
+}
+
+# Main script
+$docsPath = Join-Path $PWD "tools/azsdk-cli/docs/mcp-tools.md"
+
+if (-not (Test-Path $AzsdkCliPath)) {
+    Write-Error "Azure SDK CLI tool not found at: $AzsdkCliPath"
+    exit 1
+}
+
+$mcpProcess = $null
+
+try {
+    $version = Get-Version -CsprojPath $csprojPath
+    $mcpProcess = Start-McpServer -ExePath $AzsdkCliPath
+    $tools = Get-McpTools -Process $mcpProcess
+    $markdown = New-ToolsMarkdown -Tools $tools -Version $version
```

Comment:
Can you check that tools list is not empty to avoid creating an empty list markdown if server sends an empty list for some reason(like server took time to start the MCP)

Reply:
One additional change is to wait another 5 seconds and retry to get the tools list of tool list is not available in first attempt.